go/mcap/slicemap/slicemap.go

```diff
@@ -0,0 +1,38 @@
+// slicemap implements a generic array-map for uint16 keys and arbitrary structs.
+package slicemap
+
+import "math"
+
+// gets the item at idx, returning nil if not found.
+func GetAt[T any](items []*T, idx uint16) *T {
```

Comment:
nitpick - but if we are exposing a map-style interface, shouldn't we call these Get(k) and Put(k, v) or something? Seems like the caller should get a map abstraction and not need to understand what "At" means.

Related to that, would it be more convenient to package this as a type rather than passing the arrays in? i.e

```
 type smap[T any] []*T

func (m smap[T]) Get(key uint16) *T {
}

func (m smap[T]) Set(key uint16, value *T) {
}

func (m smap[T]) Slice() []*T {
	return m
}

func NewSmap[T any]() smap[T] {
	return make([]*T, 0)
}
```